brambling/utils/model_tables.py

```diff
@@ -482,8 +485,13 @@ def order_placed_by(self, obj):
             return "{} ({})".format(person.get_full_name(), person.email)
         return obj.order.email
 
-    def order_balance(self, obj):
-        return format_money(obj.order_balance or 0, self.event.currency)
+    def paid(self, obj):
+        paid = 0
+        for item in obj.items:
+            paid += item.price
+            for discount in item.discounts.all():
+                paid -= discount.savings()
+        return format_money(paid, self.event.currency)
```

Comment:
So the trick here is that `item.discounts.all()` is [prefetched](https://github.com/littleweaver/django-brambling/pull/634/files#diff-64a8f85cd1df4eacbf6fca6b934ed4bfR455), so it's only one db query for the whole table. I don't *think* there's really a better way to iterate over them... but I could be missing something.

From a Django perspective, `obj.items.values_list('discounts', flat=True)` doesn't work because discounts itself goes to multiple related objects. You could try annotating the items queryset with a sum of the discount amounts, then doing values_list on *that* – I think that would work. But at that point you'd be doing one query per row, rather than one per table, which would probably be less efficient.

Separately, it would be a tricky thing to calculate how much savings a discount would give someone from a query, since the amount can either be a dollar amount or a percentage.

Hmm. Actually, if we cached the real savings amount in a field instead of re-calculating it every time, we *could* annotate the items qs with total savings, probably. But the efficiency should already be pretty good as is.